v2/metadata/client.go

```diff
@@ -768,6 +788,77 @@ func (c *Client) RecordArtifact(ctx context.Context, outputName, schema string,
 	}, nil
 }
 
+func (c *Client) GetOrInsertArtifactType(ctx context.Context, schema string) (typeID int64, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("getOrInsertArtifactType(schema=%q) failed: %w", schema, err)
+		}
+	}()
+	at, err := SchemaToArtifactType(schema)
+	if err != nil {
+		return 0, err
+	}
+	getTypesRes, err := c.svc.GetArtifactType(ctx, &pb.GetArtifactTypeRequest{TypeName: at.Name})
+	if err != nil {
+		return 0, err
+	}
+	if getTypesRes.GetArtifactType() != nil {
+		return getTypesRes.GetArtifactType().GetId(), nil
+	}
+	putTypeRes, err := c.svc.PutArtifactType(ctx, &pb.PutArtifactTypeRequest{ArtifactType: at})
+	if err != nil {
+		return 0, err
+	}
+	return putTypeRes.GetTypeId(), err
+
+}
+
+func (c *Client) FindMatchedArtifact(ctx context.Context, artifactToMatch *pb.Artifact, pipelineContextId int64) (matchedArtifact *pb.Artifact, err error) {
```

Comment:
discussion: per our package responsibility architecture, `pb.Artifact` should not be exposed outside of the metadata package. metadata package should be the only package that interacts with third_party/ml_metadata data models, because it's an implementation detail.

I understand it may be challenging without a wholistic design, so I recommend:
* take some time to think about whether there's a reasonable way to do some minimal refactoring to avoid this
* if not, I'm okay with leaving a TODO comment for this PR

Reply:
The problem is the importer need to know/should know what kind of mlmd artifact it needs to import. Therefore, adding a abstract layer here to convert mlmd artifact into other artifact like runtime artifact is cumbersome

Reply:
After rethinking, maybe we can move logic related to constructing mlmd artifact from importerSpec to metadata package. But the refactor is not minimal. I will add a to do and refactor it later.